### PR TITLE
SDCICD-440: Force updateNamespace to wait for changes to occur

### PR DESCRIPTION
**Pull request by jeefy**
This ensures that `updatedNamespace` doesn't complete until the change to a namespace object has been reflected in the API. If the namespace change hasn't been reflected within 2min, it should timeout (and thus error the job)

### CODE REVIEW

File: pkg/e2e/verify/namespace_webhook.go [hunk 1]
```diff
@@ -231,7 +231,16 @@ func updateNamespace(namespace string, asUser string, userGroup string, h *helpe
 		UserName: asUser,
 		Groups:   userGroups,
 	})
-	_, err = h.Kube().CoreV1().Namespaces().Update(context.TODO(), ns, metav1.UpdateOptions{})
+	updatedNamespace, err := h.Kube().CoreV1().Namespaces().Update(context.TODO(), ns, metav1.UpdateOptions{})
+
+	err = wait.PollImmediate(2*time.Second, 1*time.Minute, func() (bool, error) {
+		ns, err = h.Kube().CoreV1().Namespaces().Get(context.TODO(), namespace, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("failed to find namespace to update '%s': %v", namespace, err)
```
Comment (cblecker) [inline pkg/e2e/verify/namespace_webhook.go:239]: nit: this error doesn't seem right. you've already updated the ns, right?

File: pkg/e2e/verify/namespace_webhook.go [hunk 2]
```diff
@@ -231,7 +231,16 @@ func updateNamespace(namespace string, asUser string, userGroup string, h *helpe
 		UserName: asUser,
 		Groups:   userGroups,
 	})
-	_, err = h.Kube().CoreV1().Namespaces().Update(context.TODO(), ns, metav1.UpdateOptions{})
+	updatedNamespace, err := h.Kube().CoreV1().Namespaces().Update(context.TODO(), ns, metav1.UpdateOptions{})
```
Comment (cblecker) [inline pkg/e2e/verify/namespace_webhook.go:234]: should this error be checked, and short circuit the get below?

File: pkg/e2e/verify/namespace_webhook.go (hunk 1 above)
Comment (jeefy) [reply pkg/e2e/verify/namespace_webhook.go]: Yep! Updated.

File: pkg/e2e/verify/namespace_webhook.go (hunk 2 above)
Comment (jeefy) [reply pkg/e2e/verify/namespace_webhook.go:236]: Absolutely! Updated.

**Pull request merged**